src/lexer/lex.c

```diff
@@ -1,10 +1,27 @@
 #include "lex.h"
 
+#include <ctype.h>
 #include <string.h> // memcpy
 #include <assert.h> // assert
 
 #include <util/out.h> // error reporting
 
+// Is a character in the given string?
+int in_string(char c, char s[]) {
+    for (char* d = s; *d; ++d) {
```

Comment:
You don't need to change this, but I'm just curious why you did `*d` in the middle?

Reply:
Testing for the end of the string (if the current character is not NUL)

Reply:
Omg I should have known. I've literally done that before lolll